cpp/src/arrow/io/mremap.h

```diff
@@ -0,0 +1,80 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+#ifdef _WIN32
+#include "arrow/io/mman.h"
```

Comment:
Any reason not to put this new function in mman.h?

Reply:
Since `mman.h` ist a kind of 3rd party library and deals specifically with mmap on windows, I thought it would be better to have a separate file for `mremap` functionality on all three systems. This way it's easier to update `mman.h` should the need ever arise.

Reply:
I ended up renaming the remap method to `arrow::internal::MemoryMapRemap`. Maybe I should move it to `arrow/util` or even into `arrow/util/io-utils.h`?

Reply:
`io-util.h` and `io-util.cc` sound like the right place indeed.